pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/source/TopicSchema.java

```diff
@@ -163,6 +168,43 @@ private static boolean isProtobufClass(Class<?> pojoClazz) {
         }
     }
 
+    @SuppressWarnings("unchecked")
+    private <T> Schema<T> newSchemaInstance(String topic, Class<T> clazz, String schemaTypeOrClassName, boolean input, ClassLoader classLoader) {
```

Comment:
I don't think we should duplicate the code when adding `classLoader`. It makes code maintenance become very hard. Can we try to avoid duplicating the code here? Instead, can you just improve the method below to handle loading using an additional class loader?